File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr) {
  const stack = [];

  for (let i = 0; i < arr.length; i++) {
    if (stack.length > 0 && stack[stack.length - 1] === arr[i]) {
      stack.pop();
    }
    stack.push(arr[i]);
  }

  return stack;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public Stack solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        
        for(int i = 0; i < arr.length; i++) {
            if(!stack.isEmpty() && stack.peek() == arr[i]) {
                stack.pop();
            } stack.push(arr[i]);
        }

        return stack;
    }
}